.golangci-kal.yml

```diff
@@ -0,0 +1,55 @@
+run:
+  timeout: 10m
+  go: "1.23"
+  allow-parallel-runners: true
+
+linters:
+  disable-all: true
+  enable:
+    - kal # linter for Kube API conventions
+   
+linters-settings:
+  custom:
+    kal:
+      type: "module"
+      description: KAL is the Kube-API-Linter and lints Kube like APIs based on API conventions and best practices.
+      settings:
+        linters:
+          enable:
+          - "conditions" # Ensure conditions have the correct json tags and markers.
+          - "commentstart" # Ensure comments start with the serialized version of the field name.
+          - "integers" # Ensure only int32 and int64 are used for integers.
+          - "jsontags" # Ensure every field has a json tag.
+          - "maxlength" # Ensure all strings and arrays have maximum lengths/maximum items.
+          - "nobools" # Bools do not evolve over time, should use enums instead.
```

Comment:
Not sure about this one.

Mostly because I'm not sure if every bool can be intuitively expressed as an enum instead. I think we have a lot of fields that worked quite well as a bool and we have no need/plans to evolve them

Additionally, general edge case, in CABPK / KCP we ~ inherit a significant part of our API from kubeadm. I wouldn't like to diverge from these upstream APIs to avoid this finding.

Reply:
This is certainly one of the more controversial ones I agree. We should have a wider discussion about this, and how we might handle the inherited APIs from Kubeadm, if the linter is going to pick those up and we don't want to change them at all, then this could be awkward from a linter perspective

Reply:
besides the inconveniences it might cause, I'd like to have it enable to foce us think twice. If enabled, what would be the process to allow exceptions?

Reply:
You have two options, either you `// nolint:kal` which would disable all linter checks on this line, or you can use the golangci-lint configuration to exclude a specific instance using their more targeted/complex exclusions

If we go for the `--new-from-rev`, we could also just override the check and then once it's in main, it's not going to come up again (on PRs at least)

Reply:
Ah another reason why I don't like using enums for this. It forces everyone to tolerate "unknown" values just in case the field ever evolves (which might never happen)

Reply:
On the other hand, if you do decide to expand later, you now have maybe two bools, where one of them is only allowed to be set when the other is set to a specific value, which can be awkward (we can probably bike shed on bools for a long time)

Reply:
Yeah. Interestingly I don't remember a single bool in CAPI where we hit this issue over the last years :)

I guess I would make this a case-by-case decision